From 834ffb6b06be6c3f4669f331f888471fff3e820d Mon Sep 17 00:00:00 2001
From: Julian Greilich <j.greilich@gmx.de>
Date: Tue, 2 Sep 2025 11:53:28 +0200
Subject: a11y Android: Use correct event type for value update of the
 ProgressBar

This way, TalkBack plays a sound to represent the value of the
progressbar instead of announcing the value as text.

Task-number: QTBUG-139712
Pick-to: 6.8 6.10
Change-Id: Ic9cd4f4cf212ba3cd3d392dacce1955d7cb2bf54
---
 .../qtproject/qt/android/QtAccessibilityDelegate.java | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git x/qtbase/src/android/jar/src/org/qtproject/qt/android/QtAccessibilityDelegate.java y/qtbase/src/android/jar/src/org/qtproject/qt/android/QtAccessibilityDelegate.java
index 63a1629715ad7b14598c1eb97173419e1eead73b..df10ce7067832da83aedce7717511be4479daa81 100644
--- x/qtbase/src/android/jar/src/org/qtproject/qt/android/QtAccessibilityDelegate.java
+++ y/qtbase/src/android/jar/src/org/qtproject/qt/android/QtAccessibilityDelegate.java
@@ -255,12 +255,15 @@ class QtAccessibilityDelegate extends View.AccessibilityDelegate
                 return;
             }
 
-            final AccessibilityEvent event =
-                    AccessibilityEvent.obtain(AccessibilityEvent.TYPE_ANNOUNCEMENT);
+            final CharSequence className = getNodeForVirtualViewId(viewId).getClassName();
+            final int eventType =
+                    className != null && className.equals("android.widget.ProgressBar")
+                    ? AccessibilityEvent.TYPE_WINDOW_CONTENT_CHANGED
+                    : AccessibilityEvent.TYPE_ANNOUNCEMENT;
+            final AccessibilityEvent event = AccessibilityEvent.obtain(eventType);
 
             event.setEnabled(true);
-            event.setClassName(getNodeForVirtualViewId(viewId).getClassName());
-
+            event.setClassName(className);
             event.setContentDescription(addLocaleSpan(value));
 
             if (event.getText().isEmpty() && TextUtils.isEmpty(event.getContentDescription())) {
